lib/core/src/lib/dialogs/unsaved-changes-dialog/unsaved-changes-dialog.component.html

```diff
@@ -1,29 +1,35 @@
-<h1 mat-dialog-title class="adf-unsaved-changes-dialog-title">
-    {{ 'CORE.DIALOG.UNSAVED_CHANGES.TITLE' | translate }}
+<h1 mat-dialog-title class="adf-unsaved-changes-dialog-header">
+    {{ (data?.headerText ? data.headerText : 'CORE.DIALOG.UNSAVED_CHANGES.TITLE') | translate }}
     <button
         data-automation-id="adf-unsaved-changes-dialog-close-button"
+        class="adf-unsaved-changes-dialog-header-close-button"
         mat-icon-button
         [title]="'CLOSE' | translate"
         [mat-dialog-close]="false">
         <mat-icon>close</mat-icon>
     </button>
 </h1>
-<mat-dialog-content>
-    {{ 'CORE.DIALOG.UNSAVED_CHANGES.DESCRIPTION' | translate }}
+<mat-dialog-content class="adf-unsaved-changes-dialog-content">
+    {{ (data?.descriptionText ? data.descriptionText : 'CORE.DIALOG.UNSAVED_CHANGES.DESCRIPTION') | translate }}
+    <div class="adf-unsaved-changes-dialog-content-checkbox" *ngIf="data?.checkboxText?.length">
+        <mat-checkbox data-automation-id="adf-unsaved-changes-dialog-content-checkbox"
+                      (change)="onToggleCheckboxPreferences($event)"
+        >{{ (data?.checkboxText | translate) }}</mat-checkbox>
```

Comment:
I guess we can remove the null safety check on data, since we are already checking it above in the ngIf. Otherwise it will be reevaluated everytime as well